### PR TITLE
Run Integration tests on GHA

### PR DESCRIPTION
**Pull request by pjbgf**

### CODE REVIEW

File: .github/workflows/test.yaml
Comment (HarrisonWAffel) [inline .github/workflows/test.yaml:44]: Are the `Install-Module` and `Import-Module` statements required for this to work? My understanding was that both the windows 2019 and 2022 github runners came preconfigured with docker ([2019 details](https://github.com/actions/runner-images/blob/main/images/windows/Windows2019-Readme.md#tools), [2022 details](https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#tools)). Also, the `DockerMsftProvider` [module is deprecated](https://github.com/OneGet/MicrosoftDockerProvider?tab=readme-ov-file) (But for whatever reason still seems to work).

File: tests/integration/process_test.ps1
Comment (HarrisonWAffel) [inline tests/integration/process_test.ps1:34]: Is there a reason that this is commented out and not removed entirely? I would think testing that we can deploy the nginx exe as well as confirming that the exe works as expected would be a useful test case. Are we limited from running this because we are moving towards GHA?

**Review by HarrisonWAffel — COMMENTED** [review]
A few comments, but generally looks good!

File: tests/integration/process_test.ps1
Comment (HarrisonWAffel) [reply tests/integration/process_test.ps1:34]: Okay, sounds good!

File: .github/workflows/test.yaml
Comment (pjbgf) [reply .github/workflows/test.yaml:44]: You are absolutely right, both 2019 and 2022 GHA runners come with Docker pre-installed.

The reason why I added this `Install-Module` and `Import-Module` was to be able to get [hns_test.ps1](https://github.com/rancher/wins/blob/main/tests/integration/hns_test.ps1) to run. Without this it fails with `Get-HnsNetwork` is to recognised IIRC.

Comment (HarrisonWAffel) [reply .github/workflows/test.yaml:44]: Interesting, I thought the hns utility commands would be bundled together with docker support. I guess leaving the `DockerMsftProvider` line can't hurt, if it fails to install docker then we will just use what is already installed. The `Import-Module` line makes more sense now

**Review by HarrisonWAffel — APPROVED** [review]
Looks good, but I would like to wait for @aiyengar2 to review before we merge

File: .github/workflows/test.yaml
Comment (aiyengar2) [inline .github/workflows/test.yaml:26]: don't think we need to run this step every time we run the workflow, since we know Docker will be part of the GH runner (based on Harrison's comment) we know this step should work.

File: .github/workflows/test.yaml
Comment (aiyengar2) [inline .github/workflows/test.yaml:34]: Shouldn't be necessary, it should already be installed as of https://github.com/actions/runner-images/blob/main/images/windows/Windows2019-Readme.md#language-and-runtime

**Review by aiyengar2 — CHANGES_REQUESTED** [review]

**Review by aiyengar2 — APPROVED** [review]
Will approve for now so we can get tests running, will follow up on the changes I requested in later PRs. Thanks for the contribution!

File: .github/workflows/test.yaml
Comment (pjbgf) [reply .github/workflows/test.yaml:34]: I removed this from a follow-up PR, and ended-up having issues with both windows versions:
https://github.com/rancher/wins/actions/runs/8094456813/job/22119187899?pr=159
https://github.com/rancher/wins/actions/runs/8094456813/job/22119187595?pr=159

**Pull request merged**
